### PR TITLE
Use Monaco for custom css editor

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

If/when #16287 is merged, the entity Custom UI form will be the last thing in GLPI that uses Codemirror. It doesn't really make sense to have two libraries for code editing/syntax highlighting if it can be avoided. Monaco seems to be the more complete alternative and as noted with the other PR, fulfills our requirements easier (in my opinion). Therefore, I recommend replacing Codemirror with Monaco and then dropping the Codemirror dependency completely.

The Monaco replacement offers code completion as before, as well as these new features:
- Linting. The editor now validates syntax and ensures users are typing valid CSS (not SCSS)
- Inline color preview and selector: Typing a hex color code, rgb function, etc, will put a preview/selector control before the color value. This selector also lets you switch the value between hex, rgb, hsl, etc.
- Inline MDN documentation for CSS properties. Seems bundles with Monaco and doesn't require an Internet connection.
- Information about selector specificity when hovering over selectors
- Contextual suggestions. For example, only the valid options for `flex` will be shown when changing that property's value.

This would require the other PR to be merged first. If testing this PR before the other PR is merged, make sure to check out the commit before the one that actually drops the Codemirror dependencies if you want the debug bar to work.

### CODE REVIEW

**Review by orthagh — APPROVED** [review]

File: templates/pages/admin/entity/custom_ui.html.twig [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                        $('.alert-info').addClass('d-none');
+                        $('#custom_css_container').removeClass('d-none');
+                    } else {
+                        $('.alert-info').removeClass('d-none');
+                        $('#custom_css_container').addClass('d-none');
+                    }
+                    refreshCustomCSSEditor();
+                });
+                $('#custom_css_container').closest('form').on('formdata', (e) => {
+                    e.originalEvent.formData.delete('entities_id');
```
Comment (cedric-anne) [inline templates/pages/admin/entity/custom_ui.html.twig:71]: #16331 should prevent to have to do this.
```suggestion
```

File: inc/define.php [hunk 2]
```diff
@@ -608,7 +608,7 @@
         'config' => ['clipboard'],
         'webhook' => ['monaco', 'autocomplete'],
     ],
-    'admin'        => ['clipboard'],
+    'admin'        => ['clipboard', 'monaco'],
```
[the remark this answers is not in the document]
Comment (cconard96) [reply inc/define.php:611]: Done. Sector-based loading is only done for the `js/modules/Monaco/MonacoEditor.js` and the relatively small Monaco CSS file. The main Monaco library will only be loaded at the exact time it is needed, much like the Vue code. This is done through stub functions in the MonacoEditor script which wrap the related library use with a dynamic import.

**Review by cedric-anne — APPROVED** [review]

File: templates/pages/admin/entity/custom_ui.html.twig [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+{% extends "generic_show_form.html.twig" %}
+{% import 'components/form/fields_macros.html.twig' as fields %}
+{% set params  = params ?? [] %}
+
+{% block form_fields %}
+    {% block more_fields %}
+        {{ fields.dropdownArrayField('enable_custom_css', item.fields['enable_custom_css'], enable_css_options, __('Enable CSS customization')) }}
+        <div id="custom_css_container" class="custom_css_container {{ item.fields['enable_custom_css'] == 1 ? '' : 'd-none' }}" style="height: 400px"></div>
+        <div class="alert alert-info {{ item.fields['enable_custom_css'] == 1 ? 'd-none' : '' }}">{{ __('Custom CSS is disabled') }}</div>
```
Comment (cedric-anne) [inline templates/pages/admin/entity/custom_ui.html.twig:42]: On a sub-entity, this message is displayed when the "Inherits configuration from the parent entity" is selected. This is not corresponding to the effective configuration when the parent entity has custom CSS enabled.
Is it possible to display the inherited value?

**Review by cedric-anne — CHANGES_REQUESTED** [review]
I rebased to fix conflicts, but my last remark is still valid.

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**